### ISSUE TITLE
Unify API errors on RFC7807 ProblemDetails (FE validation + Hub)

### ISSUE THREAD

**Issue by PeterMarinov** [endatix/endatix#168]
## Status

**Ship target:** API 0.7.5 (breaking for request-validation 400 shape).

| Slice | Repo | State |
| --- | --- | --- |
| Handler errors → `ProblemHttpResult` / `ToProblem` | endatix | **Done** (#998) |
| One RFC7807 body for FE FluentValidation + RFC members + unhandled 500s | endatix | **Done in tree** (PR pending) |
| Drop legacy FE `{statusCode,message,errors}` parser | endatix-hub | **Done** — stacked on SJS v3 → https://github.com/endatix/endatix-hub/pull/915 (`feature/h168-problem-details-validation`) |

### Done — API (this slice)

- Shared factory: `EndatixProblemDetails` (`type`, `title`, `status`, `detail`, `instance`, `traceId`, optional `errorCode` / `fields`).
- FastEndpoints: custom `c.Errors.ResponseBuilder` (not stock `UseProblemDetails()` — that emits `errors:[{name,reason}]`).
- `ToProblem` + Export stream errors go through the factory.
- Unhandled exceptions: `EndatixExceptionHandler` (`IExceptionHandler`); path `/error` removed; 500 body never leaks `ex.Message`.
- Tests: unit factory / `ToProblem` / exception handler; integration FE 400 create-form + 404 asserts `type`/`instance`/`traceId`.
- Agent docs: `oss/AGENTS.md`, API rules — dual-shape gap closed.

### Done — Hub (this slice)

- Removed `ValidationProblemDetailsSchema`; `parseProblemDetails` accepts RFC body only.
- Auth + mapper tests flipped to `fields` / `traceId`.
- New `__tests__/problem-details.test.ts` + `http-error-mapper.test.ts`.
- Delivery: `gh stack` on `feature/h908-sjs-v3-upgrade` (do not merge Hub layer to `main` before API 0.7.5 problem+json lands).
- Scope: `lib/endatix-api` only — not `services/api.ts` migration, not `lib/result` wire parse.

### Canonical wire (all API errors)

```json
{
  "type": "https://www.rfc-editor.org/rfc/rfc9110#name-400-bad-request",
  "title": "There was a problem with your request",
  "status": 400,
  "detail": "Name is required.",
  "instance": "/api/forms",
  "traceId": "0HMPNHL0JHL76:00000001",
  "errorCode": "NotEmptyValidator",
  "fields": { "name": ["Name is required."] }
}
```

**Breaking:** request-validation 400 is now problem+json (`fields`), not `{statusCode, message, errors}`.

### Remaining

- [ ] Open / merge **endatix** PR for the API factory + FE ResponseBuilder + tests (local `oss/` changes).
- [ ] Merge Hub #915 after h908 + API pairing (or keep stacked until both ready).
- [ ] Release notes 0.7.5: Compatibility table (Hub floor = build that drops legacy validation schema).

### Out of scope

- Custom `Results<Ok, Problems<...>>` / base Endpoint class — `TypedResultsBuilder` covers compile-time success vs `ProblemHttpResult`.
- Finishing Hub `services/api.ts` → `lib/endatix-api` migration.

Related: #998. Supersedes closed Hub PR #913 (branch rename).

---

## Goal (original)

RFC7807 / RFC9457 Problem Details for Endatix API errors (400 / 404 / 409 / 500).

## Scope (original checklist)

- [x] ProblemDetails response for handler failures
- [x] ProblemDetails for FluentValidation / FE request errors
- [x] ProblemDetails for unhandled exceptions (no sensitive leak)
- [x] Swagger / `ProducesProblem` + FE `ProducesMetadataType`
- [x] Tests (unit + integration + Hub client)
- [x] Compile-time `Results<Ok|Created, ProblemHttpResult>` (via TypedResultsBuilder)
- [ ] Base Endpoint class — **deferred**

### Suggested solution (historical)

OOB FastEndpoints `UseProblemDetails` + custom Results union — **not used**. Custom `ResponseBuilder` + `ProblemHttpResult` instead.

## Resource

- https://www.milanjovanovic.tech/blog/problem-details-for-aspnetcore-apis
- https://fast-endpoints.com/docs/configuration-settings#rfc7807-rfc9457-compatible-problem-details
- https://timdeschryver.dev/blog/translating-exceptions-into-problem-details-responses#problem-details-in-aspnet
